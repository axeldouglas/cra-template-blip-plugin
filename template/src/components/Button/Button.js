import React from 'react';
import PropTypes from 'prop-types';

const Button = ({
    variant,
    icon,
    text,
    arrow,
    disabled,
    children,
    aria_label = 'bds-button',
    data_testid = 'bds-button',
    onClick = () => {}
}) => {
    return (
        <bds-button
            data-testid={data_testid}
            aria-label={aria_label}
            icon={icon}
            variant={variant}
            arrow={arrow}
            disabled={disabled}
            onClick={onClick}
        >
            {!!text ? text : children}
        </bds-button>
    );
};

Button.propTypes = {
    variant: PropTypes.string,
    icon: PropTypes.string,
    text: PropTypes.string,
    arrow: PropTypes.bool,
    disabled: PropTypes.bool,
    children: PropTypes.any,
    aria_label: PropTypes.string,
    data_testid: PropTypes.string,
    onClick: PropTypes.func
};

export default Button;
