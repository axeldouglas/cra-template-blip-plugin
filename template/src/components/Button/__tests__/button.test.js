import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '@testing-library/react';

import Button from '../Button';

describe('Button component', () => {
    const handleClick = jest.fn();
    let rtlContainer;

    beforeEach(() => {
        const { container } = render(
            <Button
                text="Adicionar Plugin"
                icon="upload"
                variant="ghost"
                arrow={false}
                disabled={false}
                onClick={handleClick}
            />
        );
        rtlContainer = container;
    });

    it('should render correctly', () => {
        expect(screen.queryByTestId('bds-button')).toBeInTheDocument();
        expect(screen.queryByTestId('bds-button')).toHaveTextContent(
            'Adicionar Plugin'
        );
    });

    it('should handle click event', () => {
        const back_button_node = screen.queryByTestId('bds-button');
        fireEvent.click(back_button_node);
        expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('should matches with snapshot', () => {
        expect(rtlContainer.firstChild).toMatchSnapshot();
    });
});
