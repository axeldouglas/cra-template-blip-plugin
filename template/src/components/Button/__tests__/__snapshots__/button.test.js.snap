// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should matches with snapshot 1`] = `
<bds-button
  aria-label="bds-button"
  arrow="false"
  data-testid="bds-button"
  disabled="false"
  icon="upload"
  variant="ghost"
>
  Adicionar Plugin
</bds-button>
`;
